checkers/components/Game.tsx

```diff
@@ -0,0 +1,236 @@
+import React from 'react';
+
+import {StyleSheet, TouchableWithoutFeedback, View, Image, ImageBackground, Text} from 'react-native';
+import {GameModule, GameProps} from '../../common/common';
+import {localize} from '../../common/localize';
+import {useStoreContext} from '../../common/store';
+
+import {
+  IState,
+  getInitialState,
+  checkRiddleData,
+  isOwnColor,
+  getColor,
+  BoardDelta,
+  getAllPossibleMoves,
+  createMiniMove,
+} from '../gameLogic';
+import {getPossibleMoves, getStateScoreForIndex0} from '../aiService';
+import {riddleLevels} from '../riddles';
+
+let allPossibleMoves: BoardDelta[] = [];
+let selectedMovingPiece: BoardDelta;
+let pieceIsSelected = false;
+
+const styles = StyleSheet.create({
+  // See how to have a square component using aspectRatio=1
+  // https://reactnative.fun/2017/06/21/ratio/
+  fixedRatio: {
+    flex: 1,
+    aspectRatio: 1,
+  },
+  container: {
+    margin: 10,
+    flex: 1,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  boardImage: {
+    width: '100%',
+    height: '100%',
+    resizeMode: 'stretch',
+  },
+  boardRowsContainer: {
+    flex: 1,
+    flexDirection: 'column',
+    alignItems: 'stretch',
+    justifyContent: 'space-around',
+    padding: 0,
+    margin: 0,
+  },
+  boardCellsContainer: {
+    flex: 1,
+    flexDirection: 'row',
+    alignItems: 'stretch',
+    justifyContent: 'space-around',
+    padding: 0,
+    margin: 0,
+  },
+  boardCell: {
+    width: '12.5%',
+    height: '100%',
+    padding: 0,
+    margin: 0,
+  },
+  pieceImage: {
+    marginTop: '14%',
+    marginLeft: '12%',
+    width: '80%',
+    height: '80%',
+  },
+  bottomView: {
+    height: 100,
+  },
+  text: {
+    marginTop: 20,
+    fontSize: 19,
+    fontWeight: 'bold',
+    textAlign: 'center',
+  },
+});
+
+export default function getCheckersGameModule(): GameModule<IState> {
+  return {
+    gameId: 'checkers',
+    gameLocalizeId: 'CHECKERS_GAME_NAME',
+    initialState: getInitialState(),
+    component: CheckersComponent,
+    riddleLevels,
+    getPossibleMoves,
+    getStateScoreForIndex0,
+    checkRiddleData,
+  };
+}
+
+const CheckersComponent: React.FunctionComponent<GameProps<IState>> = (props: GameProps<IState>) => {
```

Comment:
This is a long config for the component, I think some comments explaining each part should be added